tests/rust/src/bin/path_open_preopen.rs

```diff
@@ -0,0 +1,171 @@
+unsafe fn find_first_preopened_fd() -> Result<wasi::Fd, String> {
```

Comment:
I think this looks more secure; however, we should probably update the [documentation](https://github.com/WebAssembly/wasi-testsuite/blob/main/doc/specification.md) to say that preopens defined in `dir` should be **the only** preopens in the runtime; runtime can choose to e.g. by default preopen current directory since the WASI spec doesn't specify that AFAIK. Alternatively, you can also pass the dir name as a CLI argument and use `fd_prestat_dir_name` to make sure the preopen found is the right one.

Reply:
Thanks, that makes sense. I'll go with the second option of checking the directory name since that seems to be what `open_scratch_directory` (used by most of the tests) does.

Reply:
Changed so that it matches based on dir name from the CLI args 👍